### ISSUE TITLE
Metrics-Server Chart Version: 6.5.3 not rendered properly

### ISSUE THREAD

**Issue by lmad4832** [bitnami/charts#19741]
### Name and Version

bitnami/metrics-server:6.5.3

### What architecture are you using?

amd64

### What steps will reproduce the bug?

`helm install support . --dry-run` 

Chart yaml: 
``` yaml
apiVersion: v2
name: support
description: A Helm chart for supporting components

type: application
version: 1.0.0
appVersion: 1.0.0
dependencies:
- name: metrics-server
  version: "6.5.3"
  repository: "https://charts.bitnami.com/bitnami"
- name: kubernetes-dashboard
  version: "^5.0.0"
  repository: "https://kubernetes.github.io/dashboard"
```

### What do you see instead?

This chart's helper function `common.tplvalues.render` is now interfering with Kubernetes Dashboard chart's `common.tplvalues.render` function. I see a new helper function is added with the new release to merge list of values with rendering called `common.tplvalues.merge` ([code](https://github.com/bitnami/charts/blame/c586293de255112ee434f353ce0b6f744100e6f8/bitnami/common/templates/_tplvalues.tpl#L30)) and that function is calling above mentioned render function (Specially to render metadata labels). 
As of my understanding this could be avoided by having unique like helper function signatures.

**Comment by zevisert** [comment]
@lmad4832 Out of curiosity, which `helm version` are you using?

**Comment by lmad4832** [comment]
@zevisert Helm version info: 
`version.BuildInfo{Version:"v3.10.1", GitCommit:"9f88ccb6aee40b9a0535fcc7efea6055e1ef72c9", GitTreeState:"clean", GoVersion:"go1.19.2"}`

**Comment by zevisert** [comment]
Thanks, I'm having issues with the recently released helm 3.13 changing how charts that use `"common.tplvales.render"` is working, and I thought maybe your issue was related

**Comment by migruiz4** [comment]
Hi @lmad4832,

Thank you for reporting this issue.

In this case, the incompatibility is caused by kubernetes-dashboard function defined here: https://github.com/kubernetes/dashboard/blob/bafe22e25d567399cb1e7b93a412d76f9747fb87/charts/helm-chart/kubernetes-dashboard/templates/_tplvalues.tpl#L21

This helper is conflicting with the `bitnami/common` helper `common.tplvales.render`, as only one helper can be defined with the same name.

IMHO, it is kubernetes-dashboard that is generating the issue by not using the `<chart-name>.foo.bar` naming convention for its helper, and I don't think we will be modifying all of our charts to adapt to it.

FYI, we are aware of other issues with helpers 'uniqueness' (https://github.com/helm/helm/issues/11561) but have not yet implemented a general solution for all the charts, only some particular cases (https://github.com/bitnami/charts/pull/17847).
This other issue affects instances of the same chart having different majors. For that case we may implement a different naming convention `<chartname>.vX.foo.bar` that may indirectly fix this issue, but at the moment it is in our backlog and we don't have an ETA or a final decision for it.
